### PR TITLE
install: refactor code

### PR DESCRIPTION
**Pull request by better0fdead**
Removed duplicate log message: "Checking existing..."  from "tarantool-ee" installation.
Moved check for already existing version.
Removed TODO and added return of error since 'tt' now has releases.

### CODE REVIEW

File: cli/install/install.go [hunk 1]
```diff
@@ -413,8 +413,7 @@ func installTt(binDir string, installCtx InstallCtx, distfiles string) error {
 	if ttVersion == "" {
 		log.Infof("Getting latest tt version..")
 		if len(versions) == 0 {
-			// TODO Remove after first tt release (must return error: no versions).
-			ttVersion = "master"
+			return fmt.Errorf("no versions was fetched")
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/install/install.go]: changed

File: cli/install/install.go [hunk 2]
```diff
@@ -438,6 +437,16 @@ func installTt(binDir string, installCtx InstallCtx, distfiles string) error {
 		}
 	}
 
+	versionStr := search.ProgramTt + version.FsSeparator + ttVersion
+	// Check if that version is already installed.
+	log.Infof("Checking existing...")
+	if checkExisting(versionStr, binDir) && !installCtx.Reinstall {
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/install/install.go]: changed

File: cli/install/install.go [hunk 3]
```diff
@@ -438,6 +437,16 @@ func installTt(binDir string, installCtx InstallCtx, distfiles string) error {
...
+		log.Infof("%s version of tt already exists, updating symlink...", versionStr)
+		err := util.CreateSymlink(versionStr, filepath.Join(binDir, "tt"), true)
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/install/install.go:429]: added

File: cli/install/install.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/install/install.go]: added test

File: cli/install/install.go [hunk 4]
```diff
@@ -979,6 +955,14 @@ func installTarantoolEE(binDir string, includeDir string, installCtx InstallCtx,
 	distfiles string, cliOpts *config.CliOpts) error {
 	var err error
 
+	// Check bin and header directories.
+	if binDir == "" {
+		return fmt.Errorf("binDir is not set, check %s", configure.ConfigName)
```
Comment (psergee) [inline cli/install/install.go:960]: A user knows nothing about what `binDir` is. Why do we print variable name here? The same is for `includeDir`.

File: cli/install/install.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/install/install.go:429]: changed

File: cli/install/install.go (hunk 4 above)
Comment (better0fdead) [reply cli/install/install.go]: changed to .yaml names

File: test/integration/install/test_install.py [hunk 5]
```diff
@@ -359,3 +359,37 @@ def test_install_tarantool_already_exists(tt_cmd, tmpdir):
...
+    )
+    install_process_rc = install_process.wait()
+    output = install_process.stdout.read()
+    assert "already exists" in output
+    assert install_process_rc == 0
+
+    bin_path = os.path.join(tt_dir, "bin")
+    expected_bin = os.path.join(bin_path, "tt_v1.1.2")
+    tarantool_bin = os.path.realpath(os.path.join(bin_path, "tt"))
+    assert tarantool_bin == expected_bin
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply test/integration/install/test_install.py:395]: added

File: cli/install/install.go [hunk 6]
```diff
@@ -352,8 +335,8 @@ func programDependenciesInstalled(program string) bool {
 	return true
 }
 
-// checkExisting checks if program is already installed in binary directory.
-func checkExisting(version string, dst string) bool {
+// isVersionInstalled checks if version of program is already installed in binary directory.
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply cli/install/install.go]: removed that function

File: test/integration/install/testdata/test_install_tt_already_exists/tt/tt.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+tt:
+  app:
+    bin_dir: "./bin"
+    inc_dir: "./inc"
```
Comment (askalt) [inline test/integration/install/testdata/test_install_tt_already_exists/tt/tt.yaml:4]: No newline at end of file.

File: cli/install/install.go [hunk 8]
```diff
@@ -865,9 +834,13 @@ func installTarantool(binDir string, incDir string, installCtx InstallCtx,
...
-			binDir, incDir, installCtx)
-		if err != nil || versionExists {
+		if util.IsRegularFile(filepath.Join(binDir, versionStr)) &&
+			util.IsDir(filepath.Join(incDir, versionStr)) {
+			log.Infof("%s version of tarantool already exists, updating symlinks...", versionStr)
+			err = changeActiveTarantoolVersion(versionStr, binDir, incDir)
+			if err == nil {
+				log.Infof("Done")
+			}
 			return err
```
Comment (askalt) [inline cli/install/install.go:844]: Seems a more usual way, up to you.
```suggestion
			err = changeActiveTarantoolVersion(versionStr, binDir, incDir)
			if err != nil {
				return err
			}
			log.Infof("Done")
			return nil
```

File: test/integration/install/testdata/test_install_tt_already_exists/tt/tt.yaml (hunk 7 above)
Comment (better0fdead) [reply test/integration/install/testdata/test_install_tt_already_exists/tt/tt.yaml]: added

File: cli/install/install.go (hunk 8 above)
Comment (better0fdead) [reply cli/install/install.go]: changed

Comment (askalt) [reply cli/install/install.go]: Please, change here too.  https://github.com/tarantool/tt/blob/4c74294dc737b746480be5a629bcbd8cca1f229c/cli/install/install.go#L421-L425

Comment (better0fdead) [reply cli/install/install.go]: thx changed

**Review by askalt — APPROVED** [review]

**Comment by askalt** [comment]
lgtm.

**Pull request merged**